### DISCUSSION TITLE
Sqid route model binding

### DISCUSSION THREAD

**Discussion by mxaGianluca** [dedoc/scramble#431] [Q&A]
Hi there, first of all thank you for creating an amazing package.

I was wondering about the following, i'm using this package: [laravel-sqids](https://github.com/red-explosion/laravel-sqids) which adds a route model binding based on a dynamic property `$model->sqid`. 
_(this means it's not visible in the migrations, so i'm guessing dedoc isn't able to infer it's type via dbal, please correct me if i'm wrong)_

And i have the following route `api/news/{item}` and in the controller the $item resolves to a model which uses the `laravel-sqids` trait.

I was checking out this code [RequestEssentialsExtension.php](https://github.com/dedoc/scramble/blob/main/src/Support/OperationExtensions/RequestEssentialsExtension.php)  and it seems that you've included a specific rule for models using the `HasUuids` trait.

What would be the recommended way for me to write a custom plugin to do ensure that sqids are correctly inferred as strings?

**Comment by mxaGianluca** [comment]
For completeness: 

I've also tried this as a workaround:

```php
use App\Models\TodoItem;

class TodoItemsController
{
    /**
     * @param string $item The todo item being updated.
     */
    public function update(Request $request, TodoItem $item)
    {
        // ...
    }
}
```

But this no longer seems to be working and conflicts with static analysis tools like Larastan.

**Reply by romalytvynenko** [reply]
@mxaGianluca this is really a valid issue, I will get to it in the future.
